src/Behat/Behat/Context/Environment/Handler/ContextEnvironmentHandler.php

```diff
@@ -184,4 +186,22 @@ private function resolveClass($class)
 
         return $class;
     }
+
+    /**
+     * Creates argument resolvers for a given environment.
+     *
+     * @param ContextEnvironment $environment
+     *
+     * @return ArgumentResolver[]
+     */
+    private function createArgumentResolvers(ContextEnvironment $environment)
+    {
+        if ($this->resolverFactory instanceof ArgumentResolver) {
+            return $this->resolverFactory->createArgumentResolvers($environment);
+        } elseif ($this->resolverFactory instanceof SuiteScopedResolverFactory) {
+            return $this->resolverFactory->generateArgumentResolvers($environment->getSuite());
+        }
```

Comment:
To prevent a less-likely developer WTF, I'd throw an exception here: as you don't have the type-hint anymore, somebody could inject so weird thing, that would be ignored and would end up to be no argument resolver. Throwing here if `$this->resolverFactory !== null` would highlight the issue.

Reply:
Another option is to create an internal class implementing `ArgumentResolverFactory` by wrapping a `SuiteScopedResolverFactory` (which is easy to do).

This class would then wrap the SuiteScopedResolverFactory into this internal adapter implementation in case the provided argument does not implement the new interface but only the old one. And then, the private property can be guaranteed to always contain a `ArgumentResolverFactory`

Reply:
@stof I like the way you think :)